Ubuntu 14.04 to 18.04 upgrade as 14.04 is EOL

hughsaunders opened the pull request:
Ubuntu 14.04 has entered the "Extended Secuirty Maintenance" phase
which means it's repos are behind a paywall. Access can be gained via
an Ubuntu Advantage Subscription from Canonical. This commit takes
the alternate route and upgrades the base iamge to Ubuntu 18.04.

The Dockerfile installs Ruby from `ppa:brightbox/ruby-ng` however that
PPA doesn't currently supply ruby2.2 for Ubuntu
18.04[1]. The documentation[2] suggests this combination is available,
so it may be a temporary problem. To work around the problem, ruby is
bumped from 2.2 to 2.3 as 2.3 is the oldest version available for
Ubuntu 18.04.

Related: conjurinc/appliance#1067

[1] launchpad.net/~brightbox/+archive/ubuntu/ruby-ng?field.series_filter=bionic
[2] brightbox.com/docs/ruby/ubuntu

Tovli commented on lib/conjur/fpm/Dockerfile:
```diff
@@ -1,14 +1,14 @@
 
 
-# Configure bundler and gem the way the ruby:2.2 Dockerfile does, as of 
+# Configure bundler and gem the way the ruby:2.2 Dockerfile does, as of
```
shouldn't this be changed to ruby 2.3 as well?

hughsaunders replied:
Good catch, will update.

hughsaunders replied:
Actually changed my mind, its not saying that ruby2.3 is installed, the comment is talking about setting up bundler in the same way as another dockerfile, so I now think it should stay as is.

Tovli replied:
I still think that since you've updated to ruby 2.3 you should take the 2.3 docker file example.
The bundler version is the same in any case

hughsaunders replied:
As far as I can tell, the [ruby:2.3 Dockerfile](https://github.com/docker-library/ruby/blob/802421922ef50cfa05c89a3c619992acf4329986/2.3/stretch/Dockerfile) doesn't contain `BUNDLER_VERSION`, so updating the comment to point to that would probably cause confusion.

Tovli replied:
look at this file.
Anyway - your choice. I don't feel it's a show stopper

https://github.com/docker-library/ruby/blob/c88f3a67da720bfa9fb1717960d90fd5db11c757/2.3/Dockerfile

hughsaunders replied:
Hah thats the commit the comment already points to. I've updated the comment to specify 2.3 and to link to the exact file/line so it's useful to future readers.

Tovli commented on lib/conjur/fpm/Dockerfile:
```diff
@@ -1,14 +1,14 @@
     apt-add-repository -y ppa:brightbox/ruby-ng && \
     apt-get update -y && \
-    apt-get install -y build-essential git libpq5 libpq-dev ruby2.2 ruby2.2-dev libffi-dev
+    apt-get install -y build-essential git libpq5 libpq-dev ruby2.3 ruby2.3-dev libffi-dev
```
was the change needed for ubuntu 18.04?

hughsaunders replied:
Yes, because ruby2.2 isn't available in 18.04.

hughsaunders commented:
Hi @Tovli thanks for the review, mind re-checking as I've addressed your comments. 

Also appologies for the blank PR message, I have added some info which should have been present previously.

Tovli commented on CHANGELOG.md:
```diff
@@ -1,3 +1,8 @@
+# 1.11.1
+
+* Update Conjur Dockerfile from 14.04 --> 18.04.
+  This also required a bump of ruby from 2.2 to 2.3 as 2.2 isn't available in 14.04
```
typo 14.04 should be 18.04

Tovli replied:
I think it is ruby-dev that's missing and not ruby itslef https://www.brightbox.com/blog/2015/01/05/ruby-2-2-0-packages-for-ubuntu/

hughsaunders replied:
Hah, so I was wrong, its not ubuntu, its a brightbox PPA, and their [documentation](https://www.brightbox.com/docs/ruby/ubuntu/) doesn't match reality. 

```
Ruby 2.2

We maintain Ruby 2.2 packages for Ubuntu Bionic, Xenial and Trusty. They include security fixes backported by us.
```

But [launchpad](https://launchpad.net/~brightbox/+archive/ubuntu/ruby-ng?field.series_filter=bionic) shows:
![Screenshot 2019-11-28 at 16 48 31](https://user-images.githubusercontent.com/253506/69822735-036c8e00-11ff-11ea-86a0-644575f357b0.png)


I'll update the messages to clarify the source of packages.
